nautobot/core/filters.py

```diff
@@ -630,15 +643,36 @@ def _generate_lookup_expression_filters(cls, filter_name, filter_field):
         if field is None:
             return magic_filters
 
+        field_type = None
+        try:
+            field_type = cls._meta.model._meta.get_field(field_name)
+            # TODO: This should not be needed, there seems to be some behavior changes with
+            # RoleFilter filters that haven't been migrated, which is beyond the scope of this change
+            # Working around it for now
```

Comment:
Would like to better understand the issue/symptom here so we can figure out the correct fix.

Reply:
https://github.com/nautobot/nautobot/actions/runs/6215156129/job/16867760641

Reply:
Ugh, I hadn't realized that `RoleFilter` and `StatusFilter` had such completely different code structures; I would have expected both or neither to show this behavior. It's *probably* incorrect for `RoleFilter` to restrict its queryset (https://github.com/nautobot/nautobot/blob/develop/nautobot/extras/filters/mixins.py#L304-L305) based on content-type (StatusFilter doesn't do this).

Reply:
>  I would have expected both or neither to show this behavior. 

That was my exact comment to John yesterday